### ISSUE TITLE
Implement more pathfinding algorithms

### ISSUE THREAD

**Issue by brean** [brean/python-pathfinding#3]
implement the [other algorithms from PathFinding.js](https://github.com/qiao/PathFinding.js/tree/master/src/finders)

- [ ] Bi-directional A*
- [ ] Iterative Deeping A Star (IDA*)
- [ ] Jump Point Search
- [ ] Orthogonal Jump Point Search
- [ ] Breadth-First-Search
- [ ] Bi-directional Breadth-First-Search

and other algorithms
- [ ] [D*](https://en.wikipedia.org/wiki/D*)
- [ ] [Any-angle path planning](https://en.wikipedia.org/wiki/Any-angle_path_planning) (maybe needs another visualization?)

**Comment by brean** [comment]
Implemented Bi-Directional A* in https://github.com/brean/python-pathfinding/commit/4c4eb450b266590b4718131543dacd76c685c6cc

**Comment by brean** [comment]
Implemented IDA* in https://github.com/brean/python-pathfinding/commit/ae1941e71ceb8927803142bea5e96120e136e371

**Comment by brean** [comment]
use 3D and nav-mash (see https://www.youtube.com/watch?v=9RRjM1yNvy4 )

**Comment by SulRash** [comment]
I have also been thinking it would be nice to have built-in support for random walks to allow people to just use this library entirely for pathfinding and any control experiments you need to run.

**Comment by brean** [comment]
@SulRash good idea, maybe even worth its own issue. Feel free to implement and provide a pull request.
